src/mlpack/tests/split_data_test.cpp

```diff
@@ -212,3 +212,114 @@ TEST_CASE("SplitLabeledDataLargerTest", "[SplitDataTest]")
 
   CheckDuplication(std::get<2>(value), std::get<3>(value));
 }
+
+/**
+ * Check that 0 test ratio results in a full train set for stratified split
+ */
+TEST_CASE("ZeroRatioStratifiedSplitData", "[SplitDataTest]")
+{
+  mat input(2, 15);
+  input.randu();
+
+  // Set the lables to 5 0s and 10 1s.
+  const Row<size_t> labels = { 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 };
+  const double test_ratio = 0;
+
+  const auto value = Split(input, labels, test_ratio, false, true);
+  REQUIRE(std::get<0>(value).n_cols == 15);
+  REQUIRE(std::get<1>(value).n_cols == 0);
+  REQUIRE(std::get<2>(value).n_cols == 15);
+  REQUIRE(std::get<3>(value).n_cols == 0);
+}
+
+/**
+ * Check that 1 test ratio results in a full test set for stratified split
+ */
+TEST_CASE("TotalRatioStratifiedSplitData", "[SplitDataTest]")
+{
+  mat input(2, 15);
+  input.randu();
+
+  // Set the lables to 5 0s and 10 1s.
+  const Row<size_t> labels = { 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 };
+  const double test_ratio = 1;
+
+  const auto value = Split(input, labels, test_ratio, false, true);
+  REQUIRE(std::get<0>(value).n_cols == 0);
+  REQUIRE(std::get<1>(value).n_cols == 15);
+  REQUIRE(std::get<2>(value).n_cols == 0);
+  REQUIRE(std::get<3>(value).n_cols == 15);
+}
+
+/**
+ * Check if data is stratified according to labels
```

Comment:
```suggestion
 * Check if data is stratified according to labels.
```

Missing stop at the end.